cabal-install/src/Distribution/Client/CmdRepl.hs

```diff
@@ -281,15 +283,25 @@ multiReplDecision ctx compiler flags =
 -- For more details on how this works, see the module
 -- "Distribution.Client.ProjectOrchestration"
 replAction :: NixStyleFlags ReplFlags -> [String] -> GlobalFlags -> IO ()
-replAction flags@NixStyleFlags{extraFlags = r@ReplFlags{..}, ..} targetStrings globalFlags =
-  withContextAndSelectors verbosity AcceptNoTargets (Just LibKind) flags targetStrings globalFlags ReplCommand $ \targetCtx ctx targetSelectors -> do
+replAction flags@NixStyleFlags{extraFlags = replFlags@ReplFlags{..}, configFlags} targetStrings globalFlags = do
+  withCtx verbosity targetStrings $ \targetCtx ctx targetSelectors -> do
     when (buildSettingOnlyDeps (buildSettings ctx)) $
       dieWithException verbosity ReplCommandDoesn'tSupport
     let projectRoot = distProjectRootDirectory $ distDirLayout ctx
         distDir = distDirectory $ distDirLayout ctx
 
     baseCtx <- case targetCtx of
-      ProjectContext -> return ctx
+      ProjectContext -> do
+        when (null targetSelectors) $ do
+          let projectFile = projectConfigProjectFile . projectConfigShared $ projectConfig ctx
+          let pkgs = projectPackages $ projectConfig ctx
+          case pkgs of
+            [pkg] ->
+              replAction flags [pkg] globalFlags
```

Comment:
> It seems that there is some awkwardness around the fact that
>
> In a project context you have to specify a target
> In other contexts you don't have to specify a target

@mpickering when there is only one package in a project and there are no given targets, I'm automatically picking the package as the target.